internal/services/paloalto/schema/destination_nat.go

```diff
@@ -0,0 +1,163 @@
+package schema
+
+import (
+	"strconv"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/paloaltonetworks/2022-08-29/firewalls"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/validate"
+	networkValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type DestinationNAT struct {
+	Name                  string                          `tfschema:"name"`
+	Protocol              string                          `tfschema:"protocol"`
+	FrontendConfiguration []FrontendEndpointConfiguration `tfschema:"frontend_config"`
+	BackendConfiguration  []BackendEndpointConfiguration  `tfschema:"backend_config"`
+}
+
+type FrontendEndpointConfiguration struct {
+	PublicIPID string `tfschema:"public_ip_address_id"`
+	Port       int    `tfschema:"port"`
+}
+
+type BackendEndpointConfiguration struct {
+	PublicIP string `tfschema:"public_ip_address"`
+	Port     int    `tfschema:"port"`
+}
+
+// DestinationNATSchema returns the schema for a Palo Alto NGFW Front End Settings
+func DestinationNATSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"name": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: nil, // TODO - Validation needed
+				},
+
+				"protocol": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice(firewalls.PossibleValuesForProtocolType(), false),
+				},
+
+				"backend_config": BackendEndpointSchema(),
+
+				"frontend_config": FrontendEndpointSchema(),
+			},
+		},
+	}
+}
+
+func FrontendEndpointSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		MaxItems: 1,
+		Optional: true,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"public_ip_address_id": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: networkValidate.PublicIpAddressID,
+				},
+
+				"port": {
+					Type:         pluginsdk.TypeInt,
+					Required:     true,
+					ValidateFunc: nil, // TODO - Need a atoi validation func for 1 - 65535
```

Comment:
this one's important since we try parsing it as an int below without checking the error